### ISSUE TITLE
ti attach does not work

### ISSUE THREAD

**Issue by azimonti** [jeffWelling/ticgit#39]
Hello,
when I try to attach a file I had the following error.

gems/TicGit-ng-1.0.2.11/lib/ticgit-ng/command.rb:30:in `const_get': no such file to load -- ticgit-ng/command/attach (LoadError)
    from /Users/Marco/.rvm/gems/ruby-1.9.2-p290/gems/TicGit-ng-1.0.2.11/lib/ticgit-ng/command.rb:30:in`get'
    from /Users/Marco/.rvm/gems/ruby-1.9.2-p290/gems/TicGit-ng-1.0.2.11/lib/ticgit-ng/cli.rb:35:in `execute!'
    from /Users/Marco/.rvm/gems/ruby-1.9.2-p290/gems/TicGit-ng-1.0.2.11/lib/ticgit-ng/cli.rb:9:in`execute'
    from /Users/Marco/.rvm/gems/ruby-1.9.2-p290/gems/TicGit-ng-1.0.2.11/bin/ti:13:in `<top (required)>'
    from /Users/Marco/.rvm/gems/ruby-1.9.2-p290/bin/ti:19:in`load'
    from /Users/Marco/.rvm/gems/ruby-1.9.2-p290/bin/ti:19:in `<main>'

It seems normal to me, because the command is registered,
https://github.com/jeffWelling/ticgit/blob/deving/lib/ticgit-ng/command.rb

  register 'Attach', 'Attach file to ticket', 'attach'

But then the file attach.rb is not present.

thanks and regards,
Marco

**Comment by jeffWelling** [comment]
Hmm... most peculiar.  I don't remember that file ever being there or ever having used attach myself, but since I only picked up the project after the original author abandoned it I can't say for sure if it was just accidentally deleted or simply never written.  

Never-the-less, attach is something that you expect (or, at least, _I_ would expect) a bug tracker to have, so I will fix this just as soon as I finish fixing the current bug I'm on (implementing 'ti init').   

Thanks for the feedback! :)

**Comment by azimonti** [comment]
It would be nice that the webinterface will show these attachements as hyperlinks, so that they can be easily opened from the browser with a click.
thanks.

**Comment by jeffWelling** [comment]
Hello, I did some digging and it seems like the attach command has only ever been partially written.  From what I can understand from the repository history, the original author partially created it and wrote the documentation parts, but then he abandoned the project and someone else took over for awhile and that new person reorganized a bunch of code. I suspect that during the reorganization he/she just wasn't aware that the attach documentation wasn't ready for use yet and he included it in his reworking of the source code.  
So now, the end result is that we have documentation saying TicGit supports 'attach', and some of the stub code is already written, but there are large swathes of code missing that actually do the attaching.  In one commit I saw signs that a TicGit::Attachment class was used, but I have yet to find a definition for that class.  

I suspect I will have to rewrite the attachment system myself, so this won't be quick fix because I have a lot of other real-world commitments at the moment, but I will get on this as quickly as I can.  Thank you for bringing this comedy of errors to my attention :)
Jeff Welling
